bioimageio/spec/latest/build_spec.py

```diff
@@ -412,3 +425,19 @@ def build_spec(
     model = spec.schema.Model().load(serialized)
 
     return model
+
+
+def add_weights(model, weight_uri: str, root: Optional[str] = None, weight_type: Optional[str] = None, **weight_kwargs):
+    """Add weight entry to bioimage.io model."""
+    new_weights = _get_weights(weight_uri, weight_type, None, root, **weight_kwargs)[0]
+    model.weights.update(new_weights)
+
+    serialized = spec.schema.Model().dump(model)
+    model = spec.schema.Model().load(serialized)
```

Comment:
instead of this roundtrip in `add_weights` we could either do the roundtrip for the WeightEntry in `_get_weights`... 
(or here for each weight entry in the update)

Reply:
Is there a reason not to do the full Roundtrip here? But I am fine with doing the roundtrip just for the weights, can you go ahead and change it?